src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -23,28 +23,37 @@
 
 namespace mlpack {
 namespace ann /** Artifical Neural Network.  */ {
-template<typename Model, typename InitializationRuleType, class Noise>
-GAN<Model, InitializationRuleType, Noise>::GAN(
+template<
+  typename Model,
+  typename InitializationRuleType,
+  typename Noise,
+  typename PolicyType
+>
+GAN<Model, InitializationRuleType, Noise, PolicyType>::GAN(
     arma::mat& predictors,
-    Model& generator,
-    Model& discriminator,
+    Model generator,
+    Model discriminator,
     InitializationRuleType initializeRule,
     Noise noiseFunction,
-    size_t noiseDim,
-    size_t batchSize,
-    size_t generatorUpdateStep,
+    const size_t noiseDim,
+    const size_t batchSize,
+    const size_t generatorUpdateStep,
     size_t preTrainSize,
-    double multiplier):
+    const double multiplier,
+    const double clippingParameter,
+    const double lambda):
     predictors(predictors),
-    generator(generator),
-    discriminator(discriminator),
+    generator(std::move(generator)),
+    discriminator(std::move(discriminator)),
```

Comment:
@zoq I'm not sure this implementation is correct. Might be the cause of the failure?